src/app/ApplicationTemplate.Tests.Functional/KillSwitch/KillSwitchShould.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System.Linq;
+using System.Reactive.Linq;
+using System.Threading.Tasks;
+using System.Reactive.Subjects;
+using System.Threading;
+using ApplicationTemplate.DataAccess;
+using ApplicationTemplate.Tests;
+using Microsoft.Extensions.Hosting;
+using NSubstitute;
+
+namespace KillSwitch;
+
+/// <summary>
+/// Tests for the kill switch flow.
+/// </summary>
+public class KillSwitchShould : FunctionalTestBase
+{
+	private IKillSwitchRepository _killSwitchRepository;
+	private Subject<bool> _killSwitchActivatedSubject = new Subject<bool>();
+
+	/// <summary>
+	/// Tests that the kill switch page is shown when the kill switch is activated.
+	/// </summary>
+	[Fact]
+	public async Task NavigateToKillSwitchPageWhenTheKillSwitchIsActivated()
+	{
+		// Arrange
+		var vm = await this.ReachLoginPage();
+		var sectionNavigator = GetService<ISectionsNavigator>();
+
+		// Act
+		_killSwitchActivatedSubject.OnNext(true);
+
+		var timeoutTask = Task.Delay(1000);
+
+		// Waits for the navigation to be completed, we need the StartWith in case the navigation already finished before we started observing.
+		var navTask = sectionNavigator.ObserveCurrentState()
+			.Where(x => x.LastRequestState != NavigatorRequestState.Processing
+				// This sometimes takes longer, so we need to need to make sure 
+				&& x.GetCurrentOrNextViewModelType() == typeof(KillSwitchPageViewModel))
+			// The CancellationToken is so that this returns a task.
+			.FirstAsync(CancellationToken.None);
+
+		await Task.WhenAny(navTask, timeoutTask);
+
+		// Assert
+		ActiveViewModel.Should().BeOfType<KillSwitchPageViewModel>();
+	}
+
+	/// <summary>
+	/// Tests that you cannot back out of the the kill switch page when the kill switch is activated.
+	/// </summary>
+	[Fact]
+	public async Task DisableTheBackButton()
+	{
+		// Arrange
+		var vm = await this.ReachLoginPage();
+		var sectionNavigator = GetService<ISectionsNavigator>();
+
+		// Act
+		_killSwitchActivatedSubject.OnNext(true);
+
+		var timeoutTask = Task.Delay(1000);
+
+		// Waits for the navigation to be completed, we need the StartWith in case the navigation already finished before we started observing.
+		var navTask = sectionNavigator.ObserveCurrentState()
```

Comment:
It seems the same code to wait for the navigation is repeated in several tests. I suggest you extract it in a method.